doc/HOWTO/documenting-functions-and-macros.md

```diff
@@ -0,0 +1,175 @@
+```
+
+and you'll want to document this.
+
+So, create a new .pod file named doc/man3/FUNCTION.pod.
+
+If you are asked to document several related functions in that file,
+you can create a single pod file in which you document them together.
+In this case, use the name of the first function as the file name,
+like for the above example:
+
+```text
+doc/man3/BIO_set_dgram_origin.pod
+```
+
+If you do use an unrelated name (like BIO_dgram.pod) then you'll get
+a warning about that.
+
+Next, you need to add your new file to the `test/build.info` file.
+You can do this by running:
+
+```shell
+make generate_doc_buildinfo
+```
+
+this will update doc/build.info.
+You should git add the result as `generate_doc_buildinfo` is not run on every build.
+
+With these two changes, running `make doc-nits` locally should
+now agree with you that you have documented all your new defines,
+but it might then complain:
+
+```text
+BIO_get_dgram_dest(3) is supposedly internal
+(maybe missing from other.syms) but is documented as public
+```
+
+If it is the case that your interface is meant to be public, then you need
+to edit the file `util/other.syms` to add the names of your `#define`
+functions.
+This file gets sorted alphabetically prior to each major release,
+but new additions should be placed at the end of the file.
+
+### Example
+
+For demonstration purposes, two new public symbols have been added
+by "implementing" a public function `BIO_set_dgram_foo()`
+and a public function-like macro `BIO_set_dgram_bar()`:
+
+```diff
+diff --git a/crypto/bio/bss_dgram.c b/crypto/bio/bss_dgram.c
+index 82d382cf4e..30382f0abe 100644
+--- a/crypto/bio/bss_dgram.c
++++ b/crypto/bio/bss_dgram.c
+@@ -192,6 +192,13 @@ BIO *BIO_new_dgram(int fd, int close_flag)
+     return ret;
+ }
+
++
++int BIO_set_dgram_foo(BIO* b, int foo)
++{
++    return foo;
++}
++
++
+ static int dgram_new(BIO *bi)
+ {
+     bio_dgram_data *data = OPENSSL_zalloc(sizeof(*data));
+diff --git a/include/openssl/bio.h.in b/include/openssl/bio.h.in
+index c70185db34..4ddea2f96b 100644
+--- a/include/openssl/bio.h.in
++++ b/include/openssl/bio.h.in
+@@ -485,6 +485,9 @@ struct bio_dgram_sctp_prinfo {
+ #define BIO_set_dgram_dest(b, addr)   BIO_ctrl(b, BIO_CTRL_DGRAM_SET_PEER, 0, addr)
+ #define BIO_get_dgram_dest(b, addr)   BIO_ctrl(b, BIO_CTRL_DGRAM_GET_PEER, 0, addr)
+
+ +int BIO_set_dgram_foo(BIO* b, int foo);
+ +#define BIO_set_dgram_bar(b, bar) BIO_ctrl(b, BIO_CTRL_DGRAM_SET_ADDR, 0, bar)
+ +
+  /*
+   * name is cast to lose const, but might be better to route through a
+   * function so we can do it safely
+```
+
+If you run `make doc-nits`, you might be surprised that it only
+complains about the undocumented macro, not the function:
+
+```
+~/src/openssl/master$ make doc-nits
+
+/usr/bin/perl ./util/find-doc-nits -c -n -l -e
+include/openssl/bio.h: macro BIO_set_dgram_bar(3) undocumented
+# 1 macros undocumented (count is approximate)
+make: *** [Makefile:3833: doc-nits] Error 1
+```
```

Comment:
````suggestion
```console
$ make doc-nits

/usr/bin/perl ./util/find-doc-nits -c -n -l -e
include/openssl/bio.h: macro BIO_set_dgram_bar(3) undocumented
# 1 macros undocumented (count is approximate)
make: *** [Makefile:3833: doc-nits] Error 1
```
````